controller_manager:
  ros__parameters:
    update_rate: 30  # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    omni_car_controller:
      type: omni_car_controller/RobotController

omni_car_controller:
  ros__parameters:
    joints:  # [front_left, front_right, rear_left, rear_right]
      - wheel_front_left_joint
      - wheel_front_right_joint
      - wheel_rear_left_joint
      - wheel_rear_right_joint

    command_interfaces:
      - velocity

    state_interfaces:
      - velocity

    wheel_radius: 0.06  # Radius of the wheel in meters
    wheel_base: 0.15    # Distance between left and right wheels
    wheel_track: 0.2    # Distance between front and back wheels
